tests/tests.rs

```diff
@@ -0,0 +1,254 @@
+#[cfg(test)]
+mod tests {
+	use mockito::{self, Matcher};
+	use prost::Message;
+	use std::future::Future;
+	use vss_client::client::VssClient;
+	use vss_client::error::VssError;
+
+	use vss_client::types::{
+		ErrorCode, ErrorResponse, GetObjectRequest, GetObjectResponse, KeyValue, ListKeyVersionsRequest,
+		ListKeyVersionsResponse, PutObjectRequest, PutObjectResponse,
+	};
+
+	const GET_OBJECT_ENDPOINT: &'static str = "/getObject";
+	const PUT_OBJECT_ENDPOINT: &'static str = "/putObjects";
+	const LIST_KEY_VERSIONS_ENDPOINT: &'static str = "/listKeyVersions";
+
+	#[tokio::test]
+	async fn test_get() {
+		// Spin-up mock server with mock response for given request.
+		let base_url = mockito::server_url().to_string();
+
+		// Set up the mock request/response.
+		let get_request = build_get_request();
```

Comment:
It would be better to inline the request here so the test is self-contained. Otherwise, you have to jump to the end of the file to see how this relates to the response.